src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala

```diff
@@ -59,15 +63,20 @@ final case class PurposeVersionActivation(
       waitingForApprovalVersion <- purposeManagementService.waitForApprovalPurposeVersion(
         purpose.id,
         draftVersion.id,
-        StateChangeDetails(ChangedBy.CONSUMER)
+        StateChangeDetails(ChangedBy.CONSUMER, Some(dateTimeSupplier.get()))
       )
     } yield waitingForApprovalVersion
 
     def activate(version: PurposeVersion, changedBy: ChangedBy): Future[PurposeVersion] = {
+      val newSuspendedAt =
+        if (purpose.suspendedByConsumer.contains(true) && purpose.suspendedByProducer.contains(true))
```

Comment:
It's not clear to me why we are setting here a new `suspendedAt`.
We are activating a version, so why a new value for this date?

Reply:
> It's not clear to me why we are setting here a new `suspendedAt`. We are activating a version, so why a new value for this date?

It was necessary because whenever a purpose is reactivated after being suspended we need to set the `suspendedAt` at `None`, meanwhile if a Producer or a Consumer is trying to activate it while it's suspended by the other one, it should remain suspended with the `suspendedAt` value it already had.

Now i'm implementing [this change ](https://github.com/pagopa/interop-be-purpose-management/pull/197#discussion_r1163763762) that will able us to remove this condition by passing simply the timestamp.